rust-executor/src/graphql/mutation_resolvers.rs

```diff
@@ -104,34 +107,72 @@ impl Mutation {
         passphrase: String,
     ) -> FieldResult<AgentStatus> {
         check_capability(&context.capabilities, &AGENT_CREATE_CAPABILITY)?;
+        let agent_instance = AgentService::instance();
+        {
+            let mut agent_service = agent_instance.lock().expect("agent lock");
+            let agent_ref: &mut AgentService = agent_service.as_mut().expect("agent instance");
+
+            agent_ref.create_new_keys();
+            agent_ref.save(passphrase.clone());
+        }
+
         let mut js = context.js_handle.clone();
         let script = format!(
             r#"JSON.stringify(
                 await core.callResolver("Mutation", "agentGenerate", {{ passphrase: "{}" }})
             )"#,
             passphrase
         );
-        let result = js.execute(script).await?;
-        let result: JsResultType<AgentStatus> = serde_json::from_str(&result)?;
-        result.get_graphql_result()
+        js.execute(script).await?;
+
+        let agent = {
+            let mut agent_service = agent_instance.lock().expect("agent lock");
+            let agent_ref: &mut AgentService = agent_service.as_mut().expect("agent instance");
+            agent_ref.dump().clone()
+        };
```

Comment:
Seeing these blocks with boilerplate mutex locking code, I think `AgentService` (like all our singleton objects) would benefit from having a `with_global_instance()` function like [the one in Ad4mDb](https://github.com/coasys/ad4m/blob/d75127f7a31328a9a8f14863a29e6128f04979d0/rust-executor/src/db.rs#L648) so we both have more readable code without the risk of creating deadlocks by keeping locks for too long..